src/platforms/platform.js

```diff
@@ -4,63 +4,101 @@ import helpers from '../helpers/index';
 import basic from './platform.basic';
 import dom from './platform.dom';
 
-// @TODO Make possible to select another platform at build time.
-const implementation = dom._enabled ? dom : basic;
+function extendPlatform(implementation) {
```

Comment:
This seems a bit odd to me. I know you're mostly following what's in the code currently, but I'm wondering if while we're working on this section we shouldn't clean it up a bit rather than exposing this new API. Can we use ES6 classes instead? 

Might need to move this base platform into a new file to avoid creating a cycle. Then in the platform files you can extend it. E.g. in platform.basic.js:

```
export default class BasicPlatform extends BasePlatform {
...
}
```

Reply:
Makes sense, I'll be more liberal with changing around what exists now.

Reply:
Agreed, I think this would be a good change